mrp_subcontracting_skip_no_negative/tests/test_mrp_subcontracting_skip_no_negative.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.exceptions import ValidationError
+from odoo.tests import Form, common
+
+
+class TestMrpSubcontractingSkipNoNegative(common.TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.picking_type_id = cls.env.ref("stock.picking_type_in")
+        cls.subcontractor_partner = cls.env["res.partner"].create(
+            {
+                "name": "subcontractor_partner",
+                "company_id": cls.env.ref("base.main_company").id,
+            }
+        )
+        # Create a BOM of subcontracting type
+        cls.comp = cls.env["product.product"].create(
+            {
+                "name": "Component",
+                "type": "product",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+        cls.finished = cls.env["product.product"].create(
+            {
+                "name": "finished",
+                "type": "product",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+        bom_form = Form(cls.env["mrp.bom"])
+        bom_form.type = "subcontract"
+        bom_form.product_tmpl_id = cls.finished.product_tmpl_id
+        bom_form.subcontractor_ids.add(cls.subcontractor_partner)
+        with bom_form.bom_line_ids.new() as bom_line:
+            bom_line.product_id = cls.comp
+            bom_line.product_qty = 1
+
+        cls.bom = bom_form.save()
+
+    def test_skip_no_negative_with_no_component_stock(self):
+        picking_form = Form(self.env["stock.picking"])
+        picking_form.picking_type_id = self.env.ref("stock.picking_type_in")
+        picking_form.partner_id = self.subcontractor_partner
+        with picking_form.move_ids_without_package.new() as move:
+            move.product_id = self.finished
+            move.product_uom_qty = 1
+        subcontracting_receipt = picking_form.save()
+        subcontracting_receipt = subcontracting_receipt.with_context(
+            test_stock_no_negative=True
+        )
+        subcontracting_receipt.action_confirm()
+        self.assertEqual(subcontracting_receipt.state, "assigned")
+        immediate_wizard = subcontracting_receipt.button_validate()
+        self.assertEqual(immediate_wizard.get("res_model"), "stock.immediate.transfer")
+        immediate_wizard_form = Form(
+            self.env[immediate_wizard["res_model"]].with_context(
+                **immediate_wizard["context"]
+            )
+        ).save()
+        with self.assertRaises(ValidationError):
+            immediate_wizard_form.process()
+
+    def test_skip_no_negative_with_component_stock(self):
```

Comment:
I think this test should be combined with the first one, instead of repeating the same operation in a separete test.